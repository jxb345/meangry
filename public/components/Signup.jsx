import React, { useState } from 'react';

function Signup () {
  const [email, setEmail] = useState({});
  const [signup, setSignup] = useState(true);


  const handleChangeForEmail = (e) => {
    setEmail({ email: e.target.value });

  }

  const handleSubmit = () => {
    console.log('emai', email)
    fetch('/email', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(email)
    })
    .then(res => {
      console.log('res', res);
    })
  }

  return (
    <div>
      <p className="receive-text">
        ...or <i>receive</i> anonymous heatMail.
      </p>
      <form>
        <input type="text" className="email-signup" placeholder="your email address" onChange={handleChangeForEmail}/>
        <br/>
        <button className="signup-button" type="button" onClick={handleSubmit}>SUBMIT</button>

    </form>
    </div>
  )

}

export default Signup;