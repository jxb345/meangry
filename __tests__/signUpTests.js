import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Signup from '../public/components/Signup.jsx';

