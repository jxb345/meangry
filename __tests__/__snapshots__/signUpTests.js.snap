// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preview-snapshot-test 1`] = `
<div>
  <p
    className="receive-text"
  >
    ...
    <i>
      receive
    </i>
     anonymous email:
  </p>
  <form>
    <input
      className="email-signup"
      onChange={[Function]}
      placeholder="your email address"
      required={true}
      type="text"
    />
    <br />
    <button
      className="signup-button"
      id="signup"
      onClick={[Function]}
      type="button"
    >
      Sign Up
    </button>
  </form>
  <div>
    <br />
    <br />
  </div>
  <div
    className="signup-tally"
  >
    0
     signed up to receive heatMail.
  </div>
</div>
`;
